CIAC-12379 - Align Zoom Pack docs (#38044)
demisto/content: Packs/Zoom/Integrations/Zoom/README.md

## Create Zoom ChatBOT app
1. Log onto the Zoom App Marketplace https://marketplace.zoom.us/.
2. Click **Develop** > **Build App**.
3. Choose **General App**.
4. Click **Create**.

> ![enter image description here](https://github.com/demisto/content-assets/raw/master/Assets/Zoom/create-team-chat-app.gif)

Author: Yes I will update the GIFs.

## Create Zoom ChatBOT app
1. Log onto the Zoom App Marketplace: https://marketplace.zoom.us/.
2. Click **Develop** > **Build App**.
3. Choose **General App**.
4. Click **Create**.

![enter image description here](https://github.com/demisto/content-assets/raw/master/Assets/Zoom/create-team-chat-app.gif)

### Configure App Settings

> 1. Update your app’s name. By default, the build flow generates a generic name for your app. To change your app's name, select the edit icon (pencil) and update the name. To save your changes, click anywhere outside of the app name field.

Author: Won't be changes since It matches the zoom [docs](https://developers.zoom.us/docs/build-flow/create-oauth-apps/).

   - For Cortex XSOAR 8.x / XSIAM you need to run using external engine: `https://<Engine Url>:<port>`. For example, https://my-engine-url:7001. 

![enter image description here](https://github.com/demisto/content-assets/raw/master/Assets/Zoom/bot_endpoint_url.gif)

5. Click **Scopes** > **+ Add Scopes** to add the following scope permissions.

   | Scope Type | Scope Name | 
   | --- |  --- | 
   | Team Chat | /imchat:bot |
   | Team Chat | /team_chat:read:user_message:admin |
   | Team Chat | /team_chat:read:list_user_sessions:admin |
   | Team Chat | /team_chat:read:list_user_messages:admin |
   | Team Chat | /team_chat:write:user_channel:admin |
   | Team Chat | /team_chat:write:members:admin |
   | Team Chat | /team_chat:update:user_channel:admin |
   | User | /user:read:user:admin |
![enter image description here](../../doc_files/scope-premissions.png)

6. Click **Local Test** > **Add App Now** to test your app and authorize your Cortex XSOAR app.
   
7. **If mirroring is enabled in the integration configuration or using ZoomAsk**:
   1.  **Endpoint URL Requirements:**
   

>    According to the [Zoom requirements](https://developers.zoom.us/docs/api/webhooks/#webhook-endpoint-requirements) in order to receive webhooks, the Event notification endpoint URL that you specify for each event subscription must:
> 
>           - Be a publicly accessible https endpoint url that supports TLSv1.2+ with a valid certificate chain issued by a Certificate Authority (CA).
> 
>           - Employ the `https://` scheme.

Reviewer: ```suggestion
      According to the [Zoom requirements](https://developers.zoom.us/docs/api/webhooks/#webhook-endpoint-requirements) in order to receive webhooks, the Event notification endpoint URL that you specify for each event subscription must:

      - Be a publicly accessible https endpoint url that supports TLSv1.2+ with a valid certificate chain issued by a Certificate Authority (CA).

      - Employ the `https://` scheme.
```

          - Be able to accept HTTP POST requests containing JSON payloads.

>           - Be a fully qualified domain name (FQDN).
> 
>           - be able to respond with an HTTP Status Code.
> 
>     2. **Zoom Configurations:**

Reviewer: ```suggestion
      - Be a fully qualified domain name (FQDN).

      - be able to respond with an HTTP Status Code.

    - **Zoom Configurations:**
```

   | Scope Type | Scope Name | 
   | --- |  --- | 
   | Team Chat |  Enable Chatbot within Zoom Team Chat Client /imchat:bot |
   | Team Chat |  View a user's chat message /team_chat:read:user_message:admin |
   | Team Chat |  View a user's chat sessions /team_chat:read:list_user_sessions:admin |
   | Team Chat |  View a user's chat messages /team_chat:read:list_user_messages:admin |
   | Team Chat |  Create a chat channel for a user /team_chat:write:user_channel:admin |
   | Team Chat |  Add members to a chat channel /team_chat:write:members:admin |
   | Team Chat |  Update a user's chat channel /team_chat:update:user_channel:admin |

>    | User |   View all user information /user:read:user:admin |

Reviewer 2: I think this is fine, but I think it would be even simpler and clearer to only include the scope name (the part after the `/`) in the "Scope Name" column, with a note at the top of the table to search for the scope name in Zoom in the search bar at the top. Otherwise it is confusing and difficult to find the correct scope, since there are many similarly named ones.

![enter image description here](../../doc_files/scope-premissions.png)

> 1. Click **Local Test** > **Add** to test your app and authorize your Cortex XSOAR app.

Reviewer 2: ```suggestion
1. Click **Local Test** > **Add App Now** to test your app and authorize your Cortex XSOAR app.
```

Reviewer 2: I think the numbering is off again here.

   
 ![enter image description here](../../doc_files/test-zoom-app.gif)

4. **If mirroring is enabled in the integration configuration or using ZoomAsk**:

**Endpoint URL Requirements:**

> &emsp; &emsp; To receive webhooks, the Event notification endpoint URL that you specify for each event subscription must:

Reviewer 2: Let's add a link here to show these requirements actually come from Zoom? https://developers.zoom.us/docs/api/webhooks/#webhook-endpoint-requirements

&emsp; &emsp; * Be a publicly accessible https endpoint url that supports TLSv1.2+ with a valid certificate chain issued by a Certificate Authority (CA).

&emsp; &emsp; * Be able to accept HTTP POST requests.

&emsp; &emsp; * Be able to respond with a 200 or 204 HTTP Status Code.

&emsp; &emsp; ---

> &emsp; &emsp; 1. Copy the **secret token** from the "Feature" page under the "Token" section and add it to the instance configuration.

Reviewer 2: ```suggestion
&emsp; &emsp; 1. Copy the **Secret Token** from the "Features" > "Access" page under the "Token" section and add it to the instance configuration.
```

    ![enter image description here](../../doc_files/zoom-token.png)

&emsp; &emsp; 2. Configure Event Subscriptions.

> &emsp; &emsp; &emsp; 1. In the "Feature" section under "Access" in the "General Features" section, enable "Event Subscriptions".

Reviewer 2: ```suggestion
&emsp; &emsp; &emsp; 1. In the "Features" section, under "Access", in the "General Features" section, enable "Event Subscription".
```

&emsp; &emsp; &emsp; 2. Click **+Add New Event Subscription**.

&emsp; &emsp; &emsp; 3. Enter the following information:

&emsp; &emsp; &emsp; &emsp; - Subscription name: Enter a name for this Event Subscription (e.g., "Send Message Sent").

&emsp; &emsp; &emsp; &emsp; - Authentication Header Option - 

> &emsp; &emsp; &emsp; &emsp; &emsp; 1. **Default Header Provided by Zoom option**- This option allows you to use a verification token provided by Zoom. Copy the **verification token** from the "Feature" page under the "Token" section and add it to the instance configuration.

Reviewer 2: ```suggestion
&emsp; &emsp; &emsp; &emsp; &emsp; 1. **Default Header Provided by Zoom option** - This option allows you to use a verification token provided by Zoom. Copy the **Verification Token** under the "Token" section and add it to the instance configuration.
```

&emsp; &emsp; &emsp; &emsp; &emsp; ![enter image description here](../../doc_files/verification.png)

> &emsp; &emsp; &emsp; &emsp; &emsp; 2. **Basic Authentication Option** you can use Basic Authentication by providing your Zoom Client ID (OAuth) and Secret ID (OAuth) as configured in the instance configuration.

Reviewer 2: ```suggestion
&emsp; &emsp; &emsp; &emsp; &emsp; 2. **Basic Authentication option** - You can use Basic Authentication by providing your Zoom Client ID (OAuth) and Secret ID (OAuth) as configured in the instance configuration.
```

&emsp; &emsp; &emsp; &emsp; &emsp; ![enter image description here](../../doc_files/authentication_header.png)

&emsp; &emsp; &emsp; &emsp; - Event notification endpoint URL: Enter the Cortex XSOAR URL of your server (`CORTEX-XSOAR-URL`/instance/execute/`INTEGRATION-INSTANCE-NAME`) where you want to receive event notifications. This URL should handle incoming event data from Zoom. Make sure it's publicly accessible.

> &emsp; &emsp; &emsp; &emsp; - Add Events: Click **+Add Events**. Under Event types, select **Chat Message** and then select **Chat message sent**.

Reviewer 2: ```suggestion
&emsp; &emsp; &emsp; &emsp; - Add Events: Click **+Add Events**. Under Event Types, select **Chat Message**, and then select **Chat Message Sent**.
```
